packages/web/src/scss/components/Grid/_tools.scss

```diff
@@ -47,3 +48,52 @@
         }
     }
 }
+
+@function suffix-breakpoint-class($name, $value) {
+    @if $value > 0 {
+        @return unquote('-' + $name);
+    }
+
+    @return '';
+}
+
+// Generates grid-column values for each breakpoint with fallbacks to the previous breakpoint.
+// Either end or span must be specified, but if both are specified, end will be used.
+@mixin item($breakpoints) {
+    $column-start: auto;
+    $column-span: auto;
+    $column-end: '';
+    $breakpoints-count: 1;
+
+    @each $breakpoint-name, $breakpoint-value in $breakpoints {
+        $breakpoints-count: $breakpoints-count + 1;
+        $suffix: suffix-breakpoint-class($breakpoint-name, $breakpoint-value);
+
+        @include breakpoint.up($breakpoint-value) {
+            $column-start: var(--grid-item-column-start#{$suffix}, $column-start);
+
+            $column-span: var(--grid-item-column-span#{$suffix}, $column-span);
+
+            @if $column-end == '' {
+                $column-end: var(--grid-item-column-end#{$suffix});
+            } @else {
+                $column-end: var(--grid-item-column-end#{$suffix}, $column-end);
+            }
+
+            .GridItem {
+                grid-column: $column-start /
+                    string.unquote(string.insert($column-end, ', span ' + $column-span, $breakpoints-count * -1));
```

Comment:
😄 This is to insert a custom property fallback before the last relevant closing bracket, right?

Two improvement ideas:

- rename `$breakpoints-count` to `$nesting-level-counter` or similar
- abstract this line of code to a well-named function, e.g. `insert-custom-property-fallback($custom-property, $fallback, $nesting-level)`